graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -642,6 +625,11 @@ class RenameSchemaTypesVisitor(Visitor):
     # "Foo" to "String"
     type_renamed_to_builtin_scalar_conflicts: Dict[str, str]
 
+    # Collects naming errors that arise from attempting to rename a builtin scalar. If
+    # type_renamings["String"] == "Foo" when there is a String field in the schema,
```

Comment:
Should this be
```suggestion
    # type_renamings["String"] == "Foo" when there is a Foo field in the schema,
```
instead? As currently described, this should be a legal renaming since there's no stated conflict on `Foo` and the `String` type exists.

Consider clarifying the comment here and above at `type_renamed_to_builtin_scalar_conflicts` to more explicitly and precisely describe what the conflict is: is the issue that `String` is a built-in and cannot be renamed, or is the problem that the renaming is illegal since it's attempting to rename to a name that already exists, or something else entirely.

Reply:
Originally when I made this change I was thinking that if `type_renamings` attempted to rename `String` when there were no `String` fields in the schema, it'd be covered by the no-op renaming error case but after re-reading I think it makes more sense just to check all the renamings up-front as before (instead of collecting them in the process of traversing the schema) since this is a rule that is really unlikely to change anytime soon and it'd be nice to know that this is illegal immediately instead of only when a schema contains a field of that builtin type